clang/lib/Format/FormatToken.h

```diff
@@ -1877,8 +1880,9 @@ struct AdditionalKeywords {
     // In Verilog the colon in a default label is optional.
     return Tok.is(TT_CaseLabelColon) ||
            (Tok.is(tok::kw_default) &&
-            !(Next && Next->isOneOf(tok::colon, tok::semi, kw_clocking, kw_iff,
-                                    kw_input, kw_output, kw_sequence)));
+            (!Next ||
+             Next->isNotOneOf(tok::colon, tok::semi, kw_clocking, kw_iff,
+                              kw_input, kw_output, kw_sequence)));
```

Comment:
For pointers, I prefer `!(Next && Next->isOneOf(A, B)` to `!Next || Next->isNotOneOf(A, B)` because the `Next &&` part is just for avoiding dereferencing a null pointer.

Reply:
I like the `!` to be more in the inner parenthesis, and `!A || B` reads also nice as that is the canonical representation of the implication. If `Next` exists, it is not one of ...

Reply:
Logically equivalent.
If you look at the diff, the implication is quite common.
But if that's the last thing, I can change it back.

Reply:
I also have no strong preference, although I probably tend toward using `&&` with NULL ptr checks.